src/science/soilsnow/cbl_snowAccum.F90

```diff
@@ -47,6 +47,8 @@ SUBROUTINE snow_accum ( dels,  canopy, met, ssnow, soil )
 
              ssnow%tgg(i,1) = ssnow%tgg(i,1) + canopy%precis(i) * CHLF               &
                   / ( REAL( ssnow%gammzz(i,1) ) + Ccswat *canopy%precis(i) )
+             ssnow%dtmlt(i,1) = ssnow%dtmlt(i,1) + canopy%precis(i) * CHLF           &
```

Comment:
This is the correct implementation of what has been discussed to date - however, there's something suspicious here under the circumstances of a shallow snow layer, `%tgg < CTRZ` and large `%precis`. At the moment all of the rain is being converted to snow and the temperature increasing past zero - at a later point this will remelt but that probably the incorrect way to do this and truly conserve energy (instead we should limit the amount of rain that can be frozen while keeping the  combined snow <=CTRZ)

Another new issue.

Reply:
can we just keep freezing quanta of snow WHILE tgg < 0, or does it evaluate the new temperature later on?

Did you add this as an issue?

Reply:
Added as #531 - I think we keep that separate from this PR as it will require a fair bit more thinking and we really need to focus on syncing the codebases first.